vegafusion-python/vegafusion/plan_resolver.py

```
@@ -0,0 +1,319 @@
+    """
+
+    plan: LogicalPlanNode | bytes
+    datasets: dict[str, Table] = field(default_factory=dict)
+    """Sidecar dict mapping table names to Arrow-compatible tables.
+    Used to resolve InlineTableProvider markers in the plan."""
+
+
+# ResolutionResult is the return type for resolve_plan / resolve_plan_proto.
+# Return an Arrow-compatible table for full execution, or a ResolvedPlan for
+# plan rewriting.
+ResolutionResult = Union[Table, ResolvedPlan]
```

Ruff (0.15.6)

[warning] Use `X | Y` for type annotations

(UP007)